packages/forms/src/UIForm/fields/NestedListView/NestedListView.component.js

```diff
@@ -0,0 +1,254 @@
+	 * @param {Object} event
+	 * @param {Object} item
+	 */
+	onExpandToggle(event, item) {
+		this.setState(state => {
+			const { items, value, searchCriteria, toggledChildren } = state;
+			const { toggleId } = item;
+
+			const newToggledChildren = toggledChildren.includes(toggleId)
+				? toggledChildren.filter(toggled => toggled !== toggleId)
+				: [...toggledChildren, toggleId];
+
+			return getItemsProps(items, value, searchCriteria, newToggledChildren);
+		});
+	}
+
+	/**
+	 * Handle checked item (parent) in the ListView change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Object } item
+	 */
+	onParentChange(event, item) {
+		this.setState(state => {
+			const { items, value, searchCriteria, toggledChildren } = state;
+			const { enum: availableOptions } = this.props.schema.schema.properties[item.key].items;
+
+			// Toggle all values
+			const itemValue = value[item.key] || [];
+			value[item.key] = itemValue.length === 0 ? availableOptions : [];
+
+			this.onChange(event, value);
+
+			return {
+				value,
+				...getItemsProps(items, value, searchCriteria, toggledChildren),
+			};
+		});
+	}
+
+	/**
+	 * Handle checked item (child) in the ListView change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Object } checked
+	 * @param { Object } parent
+	 */
+	onCheck(event, checked, parent) {
+		this.setState(state => {
+			const { items, value, searchCriteria, toggledChildren } = state;
+			const { key } = parent;
+
+			// Toggle checked value from state
+			const isParentKeyInValue = parent.key in value;
+
+			if (!isParentKeyInValue || !value[key].includes(checked.value)) {
+				// Add
+				if (!isParentKeyInValue) {
+					value[key] = [];
+				}
+				value[key].push(checked.value);
+			} else {
+				// Remove
+				value[key] = value[key].filter(storedValue => storedValue !== checked.value);
+			}
+
+			this.onChange(event, value);
+
+			return {
+				value,
+				...getItemsProps(items, value, searchCriteria, toggledChildren),
+			};
+		});
+	}
+
+	/**
+	 * Propagate a ListView value change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Object } newValue The new Value
+	 */
+	onChange(event, newValue) {
+		const payload = { schema: this.props.schema, value: newValue };
+		this.props.onChange(event, payload);
+		this.props.onFinish(event, payload);
+	}
+
+	/**
+	 * Search input change
+	 * @param { Object } event The change event
+	 * @param { Array } value
+	 */
+	onInputChange(event, item) {
+		clearTimeout(this.timerSearch);
+		this.timerSearch = setTimeout(() => {
+			this.setState(state =>
+				getItemsProps(state.items, state.value, item.value, state.toggledChildren),
+			);
+		}, 400);
+	}
+
+	/**
+	 * Search input ENTER/ESC management
+	 * @param { Object } event The keydown event
+	 */
+	onInputKeyDown(event) {
+		if (event.keyCode === keycode('enter')) {
+			event.stopPropagation();
```

Comment:
Why the propagation need to be stopped ?

Reply:
Didn't test without it, I kept it as is from what has been done in `ListView`